solr/core/src/java/org/apache/solr/servlet/RequestRateLimiter.java

```diff
@@ -40,32 +42,35 @@
   private final Semaphore borrowableSlotsPool;
 
   private final RateLimiterConfig rateLimiterConfig;
+  private final SlotMetadata guaranteedSlotMetadata;
+  private final SlotMetadata borrowedSlotMetadata;
+  private final SlotMetadata nullSlotMetadata;
```

Comment:
can this be static? pretty minor but since it doesn't do anything, it can be shared by all the instances and save some memory